app-modules/team/tests/Team/EditTeamTest.php

```diff
@@ -40,14 +40,95 @@
 use function Pest\Laravel\actingAs;
 use function Pest\Livewire\livewire;
 
+use Filament\Forms\Components\Select;
+use Filament\Tables\Actions\AttachAction;
 use AdvisingApp\Team\Filament\Resources\TeamResource;
 use AdvisingApp\Team\Filament\Resources\TeamResource\Pages\EditTeam;
+use AdvisingApp\Team\Filament\Resources\TeamResource\RelationManagers\UsersRelationManager;
 
 // Permission Tests
 
-test('EditTeam is gated with proper access control', function () {
+// test('EditTeam is gated with proper access control', function () {
+//     $user = User::factory()->create();
+//
+//     $team = Team::factory()->create();
+//
+//     actingAs($user)
+//         ->get(
+//             TeamResource::getUrl('edit', [
+//                 'record' => $team,
+//             ])
+//         )->assertForbidden();
+//
+//     livewire(EditTeam::class, [
+//         'record' => $team->getRouteKey(),
+//     ])
+//         ->assertForbidden();
+//
+//     $user->givePermissionTo('team.view-any');
+//     $user->givePermissionTo('team.*.update');
+//
+//     actingAs($user)
+//         ->get(
+//             TeamResource::getUrl('edit', [
+//                 'record' => $team,
+//             ])
+//         )->assertSuccessful();
+//
+//     // TODO: Finish these tests to ensure changes are allowed
+//     /** @var Team $request */
+//     $request = Team::factory()->make();
+//
+//     livewire(EditTeam::class, [
+//         'record' => $team->getRouteKey(),
+//     ])
+//         ->fillForm($request->toArray())
+//         ->call('save')
+//         ->assertHasNoFormErrors();
+//
+//     $team->refresh();
+//
+//     expect($team->name)->toEqual($request->name)
+//         ->and($team->description)->toEqual($request->description);
+// });
+//
+// test('Non Super Admin Users can be added to a team', function () {
+//     $user = User::factory()->create();
+//     $team = Team::factory()->has(User::factory()->count(1))->create();
+//
+//     actingAs($user)
+//         ->get(
+//             TeamResource::getUrl('edit', [
+//                 'record' => $team,
+//             ])
+//         )->assertForbidden();
+//
+//     livewire(EditTeam::class, [
+//         'record' => $team->getRouteKey(),
+//     ])->assertForbidden();
+//
+//     $user->givePermissionTo('team.view-any');
+//     $user->givePermissionTo('team.*.update');
+//
+//     actingAs($user)
+//         ->get(
+//             TeamResource::getUrl('edit', [
+//                 'record' => $team,
+//             ])
+//         )->assertSuccessful();
+//
+//     livewire(UsersRelationManager::class, [
+//         'ownerRecord' => $team,
+//         'pageClass' => EditTeam::class,
+//     ])
+//         ->callTableAction(
+//             AttachAction::class,
+//             data: ['recordId' => $user->getKey()]
+//         )->assertSuccessful();
+// });
+//
+test('Super Admin Users cannot be added to a team', function () {
```

Comment:
Though this test does seem correct in testing that the validation error is thrown when trying to add a User with the Super Admin role. The flow of getting there could be improved.

You should start with a `$user` to pass to `actingAs` and give it the proper permissions to access the Edit team page, like in the commented out tests (that we should bring back) above.
We don't need to test access to the EditTeam page in this test, because the above test should have already validated that.

Then create a second User, something like `$superAdmin`, give that Users the Super Admin role. And test trying to add that User and that you get the proper validation error.

Using the same $user to both login as, run the action, and test the validation feels like it could cause some pollution in the future.

Reply:
@Orrison 
I will improve this as you suggested. 
sorry, Because First Time I am writing test cases so this issues happen. next time I will keep this things  in mind.

Reply:
Hi @Orrison 
All the changes done as per your suggestion and also 3rd test case is done.